### PR TITLE
RSDK-2996 Release Candidate Workflows

### PR DESCRIPTION
**Pull request by Otterverse**
This updates workflows to support release candidate builds. Main changes are that each "type" of release has its own parent workflow now (main/stable/rc/pr) and sets an environment variable to match, to make it easier to do conditional logic in child workflows.

Includes is also some simplification of flow and security fixups for the license_finder flow (which was running directly on pull request!)

Manually tested against both stable and rc tags (from another branch) as well as PR's against this test branch, and pushes to this branch as "main." (SMURF tags/renames using during testing have already been reverted to simplify review.)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,12 +23,16 @@ jobs:
   appimage:
     needs: test
     uses: viamrobotics/rdk/.github/workflows/appimage.yml@main
+    with:
```
Comment (edaniels) [inline .github/workflows/main.yml:26]: is there a way to set this once in the overall workflow?

**Review by edaniels — APPROVED** [review]
LGTM but left some comments you may want to address

File: .github/workflows/main.yml (hunk 1 above)
Comment (Otterverse) [reply .github/workflows/main.yml:26]: I tried with env variables, but they don't pass through. Only way I could find was to use inputs to each workflow like this (same as we have to do with secrets.)

Comment (edaniels) [reply .github/workflows/main.yml:26]: ah no worries then!

File: go.mod [hunk 2]
```diff
@@ -162,7 +162,7 @@ require (
 	github.com/esimonov/ifshort v1.0.4 // indirect
 	github.com/ettle/strcase v0.1.1 // indirect
 	github.com/fatih/camelcase v1.0.0 // indirect
-	github.com/fatih/color v1.14.1 // indirect
```
[the remark this answers is not in the document]
Comment (edaniels) [reply go.mod:165]: ah okay

File: .github/workflows/appimage.yml [hunk 3]
```diff
@@ -106,10 +117,14 @@ jobs:
 
     - name: Test AppImage
       run: |
+        channel="${{ github.ref_name }}"
+        if [ "$channel" = "main" ]; then
+          channel="latest"
```
[the remark this answers is not in the document]
Comment (Otterverse) [reply .github/workflows/appimage.yml:122]: Added `# we call our main branch releases "latest"`

File: .github/workflows/appimage.yml [hunk 4]
```diff
@@ -68,19 +72,26 @@ jobs:
...
+      run: echo "date=`date +%F`" >> $GITHUB_OUTPUT
+
     - name: Upload Files (Testing)
       if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
       uses: google-github-actions/upload-cloud-storage@v0.10.4
       with:
         headers: "cache-control: no-cache"
         path: 'etc/packaging/appimages/deploy/'
-        destination: 'packages.viam.com/apps/viam-server/testing/appimage/${{ steps.build_test_app.outputs.date }}/${{ github.sha }}/'
+        destination: 'packages.viam.com/apps/viam-server/testing/appimage/${{ steps.build_date.outputs.date }}/${{ github.sha }}/'
```
[the remark this answers is not in the document]
Comment (Otterverse) [reply .github/workflows/appimage.yml:94]: I didn't even know about that GITHUB_STEP_SUMMARY file. Cool stuff! I've added output to provide links to any built binaries. Two paths basically, one when PR artifacts are built (as that bails early) and another for any push-based triggers (tags or main updates.) Examples below:

Test run (PR triggered)
https://github.com/viamrobotics/rdk/actions/runs/4930753109

Test run (push triggered)
https://github.com/viamrobotics/rdk/actions/runs/4930749546

**Pull request merged**